src/main/java/org/broadinstitute/hellbender/utils/haplotype/Haplotype.java

```diff
@@ -98,11 +98,21 @@ public Haplotype trim(final Locatable loc) {
             return null;
         }
 
-        // note: trimCigarByReference does not remove leading indels, while getBasesCoveringRefInterval does remove bases
-        // of leading indels.  We must remove leading indels from the Cigar manually.
+        // note: trimCigarByReference does not remove leading or trailing indels, while getBasesCoveringRefInterval does remove bases
```

Comment:
Can you add a test to HaplotypeUnitTest that asserts this works reasonably with leading insertions and deletions? You should hopefullh be able to use the constructor to build dummy haplotype objects and assert that leading/trailing indels are being destroyed correctly.